lib/calendar.ts

```diff
@@ -2283,12 +2284,9 @@ class NonIsoCalendar implements CalendarImpl {
   ): Temporal.PlainDate {
     const overflow = ES.ToTemporalOverflow(options);
     const cache = new OneObjectCache();
-    // Intentionally alphabetical
-    const fields = ES.PrepareTemporalFields(
-      fieldsParam,
-      ['day', 'era', 'eraYear', 'month', 'monthCode', 'year'],
-      ['day']
-    );
+    const fieldNames = this.fields(['day', 'month', 'monthCode', 'year']) as readonly ES.AnyTemporalKey[];
```

Comment:
> note that `AnyTemporalKey` is too broad here, because it includes time units, timeZone/offset, calendar, etc. Should narrow it to just calendar units?

Actually, the current spec sometimes passes time units down to `fields`, so `AnyTemporalKey` is probably OK for now. I did ask @ptomato if the spec should be changed to disallow time units, but that's beyond the scope of this PR.

> Instead of casting the return value here, should we change the type declaration of the fields method parameter and return value?

Given the notes above, I take back this suggestion. Let's just leave as-is, and depending on how the time units thing works out, we can always follow up with another PR to change the types.

Reply:
Thanks for following up on this. I'm going to resolve these comments and we can revisit if the spec changes.

Was that ask in an issue over on the spec repo? If not, should it be so we can track it here?

Reply:
Sorry, no issue for that, but @ptomato said in chat that he was planning to look at removing those time units as part of other ongoing work in the other repo.

Reply:
Gotcha. I'll resolve this then and presume that if this ends up changing that it will be reflected in another rebase in the future.